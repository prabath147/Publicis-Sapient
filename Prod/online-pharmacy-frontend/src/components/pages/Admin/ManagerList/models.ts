import { PageableResponse } from "../storeList/models";

export interface Manager 
{
  managerId: number;
  name: string;
  phoneNo: string;
  licenseNo: string;
  registrationDate: Date;
  approvalStatus: "APPROVED" | "PENDING" | "REJECTED";
}


export interface ManagerListResponse extends PageableResponse{
  data:Manager[]
}