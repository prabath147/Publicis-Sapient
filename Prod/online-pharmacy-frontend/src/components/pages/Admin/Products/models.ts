
export interface CategorySet {
  categoryId: number;
  categoryName: string;
}
export interface Product {
  categorySet: CategorySet[];
  description: string;
  dosageForm: string;
  imageUrl: string;
  productId: number;
  productName: string;
  productType: boolean;
  proprietaryName: string;
  quantity: number;
}

