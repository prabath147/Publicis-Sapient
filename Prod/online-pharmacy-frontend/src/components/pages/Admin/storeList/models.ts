import { Item } from "../../Manager/models";

export interface Test {
  data: Store[];
  pageNumber: number;
  pageSize: number;
  totalRecords: number;
  totalPages: number;
  isLastPage: boolean;
}


export interface Store {
  storeId: number;
  storeName: string;
  address: Address;
  createdDate: string;
  manager: Manager;
  revenue: number;
}

export interface Address {
  addressId: number;
  street: string;
  city: string;
  state: string;
  pinCode: number;
  country: string;
}

export interface Manager {
  managerId: number;
  name: string;
  phoneNo: string;
  licenseNo: string;
  registrationDate: string;
  approvalStatus: string;
}

export interface PageableResponse{
  "pageNumber":number,
  "pageSize":number,
  "totalRecords":number,
  "totalPages":number,
  "isLastPage":boolean
}

export interface getStoreInventoryResponse extends PageableResponse{
  data : Item[]
}


