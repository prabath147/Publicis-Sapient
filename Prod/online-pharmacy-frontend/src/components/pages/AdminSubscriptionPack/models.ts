export interface Subscriptionv{
    subscriptionId:number,
    subscriptionName:string,
    description:string,
    startDate:Date,
    endDate:Date,
    benefits:Benefit
    subscriptionCost:number,
    period:number,
    status:string
}

export interface Benefit{
    delivery_discount:number,
    one_day_delivery:boolean
}
export interface SubDetails{
    subscriptionName:string,
    description:string,
    startDate:Date,
    endDate:Date,
    benefits:Benefit,
    subscriptionCost:number,
    period:number
}

export interface SubRequest{
    name:string,
    description:string,
    stDate:Date,
    edDate:Date,
    deliveryDiscount:number,
    oneDayDelivery:string,
    cost:number,
    period:number
}