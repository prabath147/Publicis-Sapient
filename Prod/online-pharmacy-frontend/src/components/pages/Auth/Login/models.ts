export interface User {
  id: number;
  username: string;
  email: string;
  role: string;
}

export interface LoginRequest {
  username: string;
  password: string;
}

export interface LoginResponse {
  id: number;
  username: string;
  email: string;

  token: string;
  refreshToken: string;
}

export interface tokenStructure {
  sub: string;
  role: string;
  iat: number;
  exp: number;
  id:number;
}
