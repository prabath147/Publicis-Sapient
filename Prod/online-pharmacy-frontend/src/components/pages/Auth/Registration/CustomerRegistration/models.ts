import { AddressInterface } from './../../../../ui/forms/AddressForm';

export interface CustomerRegistrationRequest {
  username: string;
  email: string;
  password: string;
  role: ["customer"];

  detailObject: {
    fullName: string;
    mobileNumber: number;
    address: AddressInterface
  };
}

export interface CustomerRegistrationForm {
  fullName: string;
  username: string;
  mobileNumber: number;
  email: string;
  password: string;
  confirmPassword: string;
  address: AddressInterface
}
