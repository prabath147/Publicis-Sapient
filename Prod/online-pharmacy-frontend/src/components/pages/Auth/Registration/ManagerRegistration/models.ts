export interface ManagerRegistrationRequest {
  username: string;
  email: string;
  password: string;
  role: ["manager"];

  detailObject: {
    name: string;
    phoneNo: number;
    licenseNo: number;
  };
}

export interface ManagerRegistrationForm {
  username: string;
  email: string;
  password: string;
  confirmPassword: string;
  name: string;
  phoneNo: number;
  licenseNo: number;
}
