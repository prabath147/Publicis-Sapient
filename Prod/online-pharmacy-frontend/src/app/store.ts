import { Action, configureStore, ThunkAction } from "@reduxjs/toolkit";
import StoreReducer from "../components/pages/Admin/storeList/StoreSlice";
import SubscriptionReducer from "../components/pages/AdminSubscriptionPack/SubscriptionvSlice";
import UserReducer from "../components/pages/Auth/Login/UserSlice";
import CartReducer from "../components/pages/Cart/CartSlice";
import ManagerReducer from "../components/pages/Manager/ManagerSlice";
import NotificationReducer from "../components/pages/Open/notificationList/NotificationSlice";
// import OrderSlice from "../components/pages/orderDetail/OrderSlice";
// import OrderListSlice from "../components/pages/orderList/OrderListSlice";

export const store = configureStore({
  reducer: {
    storeL: StoreReducer,
    user: UserReducer,
    notification: NotificationReducer,
    viewsubscription: SubscriptionReducer,
    // addsubscription: AddSubcriptionRe,
    cart: CartReducer,
    // order: OrderSlice,
    // orderL: OrderListSlice,
    manager: ManagerReducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
