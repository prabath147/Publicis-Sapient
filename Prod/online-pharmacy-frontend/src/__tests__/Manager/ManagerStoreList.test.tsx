import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { store } from "../../app/store";
import ManagerStoreList from "../../components/pages/Manager/ManagerStoreList/ManagerStoreList";
import { StoreListPage } from "../../components/pages/Manager/models";
import mockAxios from "../../__mocks__/axios";

const MockManagerStoreList = () => {
    return <Provider store={store}>
        <BrowserRouter>
            <ManagerStoreList />
        </BrowserRouter>
    </Provider>
}

const stores: StoreListPage = {
    data: [
        {
            storeId: 101,
            storeName: "New Store",
            address: {
                addressId: 1,
                street: "New street",
                city: "New City",
                state: "New State",
                pinCode: 123456,
                country: "Country"
            },
            manager: {
                managerId: -1,
                name: "John",
                approvalStatus: "APPROVED",
                licenseNo: "1889299",
                phoneNo: "1234567890",
                registrationDate: "2022-08-16",
            },
            createdDate: "2022-09-09",
            revenue: 1000
        }
    ],
    isLastPage: true,
    pageNumber: 1,
    pageSize: 10,
    totalPages: 1,
    totalRecords: 10
}


describe("ManagerStoreList", () => {

    afterEach(() => {
        jest.resetAllMocks();
    });

    it('component created without data', () => {
        mockAxios.get.mockImplementationOnce(() =>
            Promise.reject(),
        )
        const { asFragment } = render(<MockManagerStoreList />)
        expect(asFragment()).toMatchSnapshot();
    })

    it('component created with fail event', async () => {
        mockAxios.get.mockImplementationOnce(() =>
            Promise.reject(),
        )
        const { asFragment, getByTestId } = render(<MockManagerStoreList />)

        // await waitForElementToBeRemoved(getByTestId(/Loading/i));

        expect(asFragment()).toMatchSnapshot();
    })

    it('component creatednwith store list data', async () => {
        mockAxios.get.mockImplementationOnce(() =>
            Promise.resolve({ data: stores }),
        )
        const { asFragment, getByTestId } = render(<MockManagerStoreList />)

        // await waitForElementToBeRemoved(getByTestId(/Loading/i));

        expect(asFragment()).toMatchSnapshot();
    })


})
