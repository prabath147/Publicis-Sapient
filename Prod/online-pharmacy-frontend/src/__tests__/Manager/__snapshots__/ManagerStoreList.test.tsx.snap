// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManagerStoreList component created with fail event 1`] = `
<DocumentFragment>
  <div>
    <div
      class="mantine-LoadingOverlay-root mantine-137kca9"
      data-testid="Loading"
      style="z-index: 400;"
    >
      <svg
        class="mantine-1avyp1d"
        height="36px"
        role="presentation"
        stroke="#228be6"
        style="z-index: calc(400 + 1);"
        viewBox="0 0 38 38"
        width="36px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          fill-rule="evenodd"
        >
          <g
            stroke-width="5"
            transform="translate(2.5 2.5)"
          >
            <circle
              cx="16"
              cy="16"
              r="16"
              stroke-opacity=".5"
            />
            <path
              d="M32 16c0-9.94-8.06-16-16-16"
            >
              <animatetransform
                attributeName="transform"
                dur="1s"
                from="0 16 16"
                repeatCount="indefinite"
                to="360 16 16"
                type="rotate"
              />
            </path>
          </g>
        </g>
      </svg>
      <div
        class="mantine-Overlay-root mantine-8uy2jv"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`ManagerStoreList component created without data 1`] = `
<DocumentFragment>
  <div>
    <div
      class="mantine-LoadingOverlay-root mantine-137kca9"
      data-testid="Loading"
      style="z-index: 400;"
    >
      <svg
        class="mantine-1avyp1d"
        height="36px"
        role="presentation"
        stroke="#228be6"
        style="z-index: calc(400 + 1);"
        viewBox="0 0 38 38"
        width="36px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          fill-rule="evenodd"
        >
          <g
            stroke-width="5"
            transform="translate(2.5 2.5)"
          >
            <circle
              cx="16"
              cy="16"
              r="16"
              stroke-opacity=".5"
            />
            <path
              d="M32 16c0-9.94-8.06-16-16-16"
            >
              <animatetransform
                attributeName="transform"
                dur="1s"
                from="0 16 16"
                repeatCount="indefinite"
                to="360 16 16"
                type="rotate"
              />
            </path>
          </g>
        </g>
      </svg>
      <div
        class="mantine-Overlay-root mantine-8uy2jv"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`ManagerStoreList component creatednwith store list data 1`] = `
<DocumentFragment>
  <div>
    <div
      class="mantine-LoadingOverlay-root mantine-137kca9"
      data-testid="Loading"
      style="z-index: 400;"
    >
      <svg
        class="mantine-1avyp1d"
        height="36px"
        role="presentation"
        stroke="#228be6"
        style="z-index: calc(400 + 1);"
        viewBox="0 0 38 38"
        width="36px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          fill-rule="evenodd"
        >
          <g
            stroke-width="5"
            transform="translate(2.5 2.5)"
          >
            <circle
              cx="16"
              cy="16"
              r="16"
              stroke-opacity=".5"
            />
            <path
              d="M32 16c0-9.94-8.06-16-16-16"
            >
              <animatetransform
                attributeName="transform"
                dur="1s"
                from="0 16 16"
                repeatCount="indefinite"
                to="360 16 16"
                type="rotate"
              />
            </path>
          </g>
        </g>
      </svg>
      <div
        class="mantine-Overlay-root mantine-8uy2jv"
      />
    </div>
  </div>
</DocumentFragment>
`;
